File: Hash-Based Searching/Perfect Hashing Search/PerfectHashingSearch.java
public class PerfectHashingSearch {

    private final int[] table;

    public PerfectHashingSearch() {

        table = new int[10];

        table[0] = 11;
        table[1] = 22;
        table[2] = 33;
        table[3] = 44;
        table[4] = 55;
    }

    private int perfectHash(
        int key
    ) {

        return switch(key) {

            case 11 -> 0;
            case 22 -> 1;
            case 33 -> 2;
            case 44 -> 3;
            case 55 -> 4;

            default -> -1;
        };
    }

    public boolean search(
        int key
    ) {

        int idx =
            perfectHash(key);

        if (idx == -1)
            return false;

        return table[idx] == key;
    }

    public static void main(
        String[] args
    ) {

        PerfectHashingSearch ht =
            new PerfectHashingSearch();

        System.out.println(
            "Search 33: "
            + (
                ht.search(33)
                ? "Found"
                : "Not Found"
            )
        );
    }
}

File: Hash-Based Searching/Perfect Hashing Search/PerfectHashingSearch.py
class PerfectHashingSearch:

    def __init__(self):

        self.table = [None] * 10

        self.table[0] = 11
        self.table[1] = 22
        self.table[2] = 33
        self.table[3] = 44
        self.table[4] = 55

    def perfect_hash(
        self,
        key
    ):

        mapping = {
            11: 0,
            22: 1,
            33: 2,
            44: 3,
            55: 4
        }

        return mapping.get(
            key,
            -1
        )

    def search(
        self,
        key
    ):

        idx =
            self.perfect_hash(key)

        if idx == -1:
            return False

        return (
            self.table[idx]
            == key
        )


if __name__ == "__main__":

    ht =
        PerfectHashingSearch()

    print(
        "Search 33:",
        "Found"
        if ht.search(33)
        else "Not Found"
    )
